packages/wallet/WalletManager.js

```diff
@@ -0,0 +1,421 @@
+            break;
+          }
+          default:
+            this.account = null;
+            console.log('The network has not been set.');
+        }
+        return null;
+      } catch (err) {
+        console.log('The input is not a valid private/public key.');
+        return null;
+      }
+    } else {
+      console.log('The argument must be empty, or 0,1,2,...');
+      return null;
+    }
+  }
+
+  /**
+   * A shorthand for creating multiple accounts.
+   * The accounts created will be the 0th, 1st, ... (n-1)th in the wallet
+   * @param {int} numberOfAccounts
+   */
+  async setMultipleAccounts(numberOfAccounts) {
+    for (let i = 0; i < numberOfAccounts; i++) {
+      await this.getAccount(i);
+    }
+  }
+
+  /**
+   * Set the current account.
+   * @param {number} index - The index of the account in the accountArray
+   */
+  setCurrentAccount(index) {
+    const max = this.accountArray.length - 1;
+    if (index > max) {
+      console.log(`the maximum index is${max.toString()}`);
+    } else {
+      this.account = this.accountArray[index];
+    }
+  }
+
+  /**
+   * Return the account array.
+   * The index of accounts in this array can be different from the index
+   * of the account in the wallet,
+   * depending on the sequence the accounts are created.
+   * @returns {Array} the array of accounts
+   */
+  getAccountArray() {
+    if (this.accountArray == null) {
+      console.log('The account array is empty');
+      return null;
+    }
+    return this.accountArray;
+  }
+
+  /**
+   *
+   * @param {string} password - the password
+   * @returns {ByteStringBuffer} the object containing the encrypted string
+   */
+  encryptSeed(password) {
+    // generate a random key and IV
+    // Note: a key size of 16 bytes will use AES-128, 24 => AES-192, 32 => AES-256
+    this.iv = forge.random.getBytesSync(16);
+
+    // alternatively, generate a password-based 16-byte key
+    this.salt = forge.random.getBytesSync(128);
+    const key = forge.pkcs5.pbkdf2(password, this.salt, 10, 16); // numIterations set to 10
+
+    // encrypt some bytes using CBC mode
+    // (other modes include: ECB, CFB, OFB, CTR, and GCM)
+    // Note: CBC and ECB modes use PKCS#7 padding as default
+    const cipher = forge.cipher.createCipher('AES-CBC', key);
+    cipher.start({ iv: this.iv });
+    cipher.update(forge.util.createBuffer(this.getSeed()));
+    cipher.finish();
+    const encrypted = cipher.output;
+    // outputs encrypted hex
+    // console.log(encrypted.toHex());
+    return encrypted;
+  }
+
+  /**
+   * It must be called after encryptSeed is called at least once
+   * @param {string} password - The password to decrypt
+   * @returns {string} the seed phrases if the password is correct, otherwise a random string
+   */
+  decryptSeed(password, encrypted) {
+    const key = forge.pkcs5.pbkdf2(password, this.salt, 10, 16); // numIterations set to 10
+    const decipher = forge.cipher.createDecipher('AES-CBC', key);
```

Comment:
Is there a reason why CBC mode was used? I suggest using GCM instead

Reply:
Changed to GCM encryption, and handled the decryption failure case.

Reply:
pass options for eth web 3 encryption